src/modules/esl-media/core/esl-media.ts

```diff
@@ -371,6 +374,26 @@ export class ESLMedia extends ESLBaseElement {
     this.deferredReinitialize();
   }
 
+  @listen({
+    event: ESLToggleable.prototype.SHOW_EVENT,
+    target: window
+  })
+  protected _onContainerShow(e: Event): void {
+    const {target} = e;
+    if (!this.isAcceptableTarget(target) || !this.inViewport) return;
+    if (this.autoplay) this.play();
+  }
+
+  @listen({
+    event: ESLToggleable.prototype.HIDE_EVENT,
```

Comment:
Probably we need to use string literal instead as currently, we obligate the user to have both components in the bundle.